sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs

```diff
@@ -93,5 +95,22 @@ public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action
             builder.Services.AddSingleton<ServiceBusClientFactory>();
             return builder;
         }
+
+        public static IWebJobsBuilder AddServiceBusScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
+        {
+            ServiceBusScalerProvider provider = null;
```

Comment:
Something still seems off with this pattern. For example, if the ITargetScalerProvider is requested first before the IScaleMonitorProvider, null would be returned. To avoid race conditions like this, I think you need to define a single factory and reuse in both places.

Reply:
Proposed `Lazy<>` implementation